prow/github/client.go

```diff
@@ -191,14 +194,14 @@ func (c *Client) Throttle(hourlyTokens, burst int) {
 //   An endpoint is used when all preceding endpoints have returned a conn err.
 //   This should be used when using the ghproxy GitHub proxy cache to allow
 //   this client to bypass the cache if it is temporarily unavailable.
-func NewClient(token string, bases ...string) *Client {
+func NewClient(token func() []byte, bases ...string) *Client {
 	return &Client{
 		logger: logrus.WithField("client", "github"),
 		time:   &standardTime{},
-		gqlc:   githubql.NewClient(oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token}))),
+		gqlc:   githubql.NewClient(oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(&oauth2.Token{AccessToken: string(token())}))),
```

Comment:
This reads the token only once

Reply:
I will try to find the solution to pass the agent to graphql

Reply:
Follow-up issue for this. Most likely the solution is to update the SecretAgent to implement [TokenSource](https://godoc.org/golang.org/x/oauth2#TokenSource).

Reply:
opened #8702